src/plugins/gimbal/include/plugins/gimbal/gimbal.h

```diff
@@ -84,6 +92,34 @@ class Gimbal : public PluginBase {
      */
     void set_pitch_and_yaw_async(float pitch_deg, float yaw_deg, result_callback_t callback);
 
+    /** 
+     * @brief Set gimbal mode (synchronous).
+     *
+     * This sets the desired yaw mode of a gimbal.
+     * The function will return when the command is accepted, however, it might
+     * take the gimbal longer to actually be set to the new angles.
+     *
+     * @param gimbal_mode The mode to be set. Either yaw lock or yaw follow.
+     * Yaw lock will fix the gimbal poiting to an absolute direction. 
+     * Yaw follow will point the gimbal to the vehicle heading.
+     * @return Result of request.
+     */
+    Result set_gimbal_mode(const Gimbal::GimbalMode gimbal_mode);
+
+    /** 
+     * @brief Set gimbal mode (asynchronous).
+     *
+     * This sets the desired yaw mode of a gimbal.
+     * The function will return when the command is accepted, however, it might
+     * take the gimbal longer to actually be set to the new angles.
+     *
+     * @param gimbal_mode The mode to be set. Either yaw lock or yaw follow.
+     * Yaw lock will fix the gimbal poiting to an absolute direction. 
```

Comment:
@julianoes @JonasVautherin What do you guys think of this, should I detail it more?

Reply:
That's ok. I would adapt the description of [set_pitch_and_yaw](https://github.com/mavlink/MAVSDK/blob/f0d8db27473e6db99876fe5201093a8d319c6182/src/plugins/gimbal/include/plugins/gimbal/gimbal.h#L69) and note that the yaw angle reference depends on the gimbal mode.

Reply:
I did the follow up fix for this:  #826